<launch>

	<!-- remap topics -->
	<remap from="/hexapod/front_left_body_coxa/command" to="/hexapod/AL_coxa/command"/>
	<remap from="/hexapod/front_left_coxa_femour/command" to="/hexapod/AL_femur/command"/>
	<remap from="/hexapod/front_left_femour_tibia/command" to="/hexapod/AL_tibia/command"/>
	
	<remap from="/hexapod/front_right_body_coxa/command" to="/hexapod/AR_coxa/command"/>
	<remap from="/hexapod/front_right_coxa_femour/command" to="/hexapod/AR_femur/command"/>
	<remap from="/hexapod/front_right_femour_tibia/command" to="/hexapod/AR_tibia/command"/>

	<remap from="/hexapod/middle_left_body_coxa/command" to="/hexapod/BL_coxa/command"/>
	<remap from="/hexapod/middle_left_coxa_femour/command" to="/hexapod/BL_femur/command"/>
	<remap from="/hexapod/middle_left_femour_tibia/command" to="/hexapod/BL_tibia/command"/>

	<remap from="/hexapod/middle_right_body_coxa/command" to="/hexapod/BR_coxa/command"/>
	<remap from="/hexapod/middle_right_coxa_femour/command" to="/hexapod/BR_femur/command"/>
	<remap from="/hexapod/middle_right_femour_tibia/command" to="/hexapod/BR_tibia/command"/>

	<remap from="/hexapod/rear_left_body_coxa/command" to="/hexapod/CL_coxa/command"/>
	<remap from="/hexapod/rear_left_coxa_femour/command" to="/hexapod/CL_femur/command"/>
	<remap from="/hexapod/rear_left_femour_tibia/command" to="/hexapod/CL_tibia/command"/>

	<remap from="/hexapod/rear_right_body_coxa/command" to="/hexapod/CR_coxa/command"/>
	<remap from="/hexapod/rear_right_coxa_femour/command" to="/hexapod/CR_femur/command"/>
	<remap from="/hexapod/rear_right_femour_tibia/command" to="/hexapod/CR_tibia/command"/>

        <include file="$(find hexapod_large_gazebo)/launch/gazebo_max_tf.launch" />
	<include file="$(find hexapod_remote)/launch/hexapod_remote.launch" />
	<rosparam file="$(find simple_hexapod_controller)/config/large_hexapod.yaml" command="load"/>
	<rosparam file="$(find simple_hexapod_controller)/config/gait.yaml" command="load"/>
	<node name="simple_hexapod_controller" pkg="simple_hexapod_controller" type="simple_hexapod_controller_node" output="screen"/>	

        <param name="/hexapod/parameters/dynamixel_interface" value="true"/>
        <param name="/hexapod/parameters/pose_controller/time_to_start" value="1.0"/>
        <!--param name="/hexapod/parameters/impedance_controller/impedance_control" value="false"/-->
        <param name="/hexapod/parameters/impedance_controller/impedance_input" value="joint_effort"/>
        <param name="/hexapod/parameters/impedance_controller/force_gain" value="0.1"/>
        
</launch>

